grails-app/controllers/au/org/emii/portal/WpsController.groovy

```diff
@@ -34,21 +53,36 @@ class WpsController extends RequestProxyingController {
     }
 
     def _renderExecutionStatus(execResponse) {
+
         render(
             view: 'show',
             model: [
                 job: [
                     uuid: params.uuid,
                     reportUrl: g.createLink(action: 'jobReport', absolute: true, params: params),
-                    createdTimestamp: new DateTime(String.valueOf(execResponse.Status.@creationTime)),
-                    status: execResponse.Status.children()?.first().name(),
-                    downloadTitle: execResponse.ProcessOutputs.Output.Title.text(),
+                    createdTimestamp: new DateTime(String.valueOf(execResponse.Status.@creationTime)).toString(),
+                    status: params.status,
+                    downloadTitle: "IMOS download - ${params.uuid}",
                     downloadUrl: _getProxiedDownloadUrl(execResponse.ProcessOutputs.Output.Reference.@href)
                 ]
             ]
         )
     }
 
+    def _renderExecutionFailed(execResponse) {
+        render(
+            view: 'show',
+            model: [
+                job: [
+                    uuid: params.uuid,
+                    status: "ProcessFailed",
+                    downloadTitle: "IMOS download ERROR - ${params.uuid}",
+                    errorMessage: "message"
```

Comment:
Oh. I see we're using this value to build up a message code to load from messages.properties. It looks like this is the only place that uses it so maybe passing a boolean called `showErrorMessage` might make it more clear what is going on?

Or, if you didn't want to do that, at least renaming this to be `errorMessageCode` to make it clear that it's not the message itself, but the code to load the message.